### PR TITLE
Trigger for Alsur Survey 4 and test class when Type of delivery = Produc...

### PR DESCRIPTION
**Pull request by jgomezr**
Hi Luke

This are the alsur survey 4 trigger and test class.

The trigger is on when Type of delivery = Productor - entrega IGUAL a asignación.

### CODE REVIEW

File: src/classes/TestAlsurCreateFarmerProdDeliverytrigger.cls [hunk 1]
```diff
@@ -0,0 +1,49 @@
+@isTest
```
Comment (lc0rp) [inline src/classes/TestAlsurCreateFarmerProdDeliverytrigger.cls:1]: What is the code coverage?

File: src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+			    }
+			    delivery.Is_Delivery_On_Time__c = 'Sí';
+			    delivery.Number_of_Kilos__c = foa.Number_of_Kilos__c;
+		        delivery.Number_of_Units_of_Presentation__c = foa.Number_of_Units_of_Presentation__c;
+		        delivery.Presentacion__c = foa.Presentacion__c;
+		        deliveriesToInsert.add(delivery);
+		    }
+	    }
+	}
+    if (deliveriesToInsert.size() > 90) {
```
Comment (lc0rp) [inline src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger]: Please fix this indenting. This block should be in line with the "for" block. Use consistent indentation.

File: src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger [hunk 3]
```diff
@@ -0,0 +1,25 @@
+trigger AlsurCreateFarmerProductDeliveryComplete on Farmer_Product_Delivery__c (
+	after insert) {
+	List<Farmer_Product_Delivery__c> deliveriesToInsert = new List<Farmer_Product_Delivery__c>();
+	for (Farmer_Product_Delivery__c newItem : Trigger.new) {
+		if (newItem.TypeOfDelivery__c == 'Productor - entrega IGUAL a asignación') {
+	        for(Farmer_Order_Allocation__c foa : [SELECT Number_of_Baskets__c, Delivery_Date__c, Number_of_Kilos__c, Number_of_Units_of_Presentation__c, Presentacion__c FROM Farmer_Order_Allocation__c WHERE Id =:newItem.Farmer_Order_Allocation__c]){ 
+			    Farmer_Product_Delivery__c delivery = new Farmer_Product_Delivery__c();
```
Comment (lc0rp) [inline src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger]: Use consistent indentation.

File: src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+			    delivery.Number_of_Baskets__c = foa.Number_of_baskets__c;
+			    delivery.Delivery_Date__c = foa.Delivery_Date__c;
+			    if (foa.Alsur_Farmer__c != null) {
+			    	delivery.Farmer_Code__c = foa.Alsur_Farmer__r.Farmer_Code__c;
+			    }
+			    delivery.Is_Delivery_On_Time__c = 'Sí';
+			    delivery.Number_of_Kilos__c = foa.Number_of_Kilos__c;
+		        delivery.Number_of_Units_of_Presentation__c = foa.Number_of_Units_of_Presentation__c;
```
Comment (lc0rp) [inline src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger]: Use consistent indentation.

**Comment by jgomezr** [comment]
47%

File: src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger (hunk 2 above)
Comment (lc0rp) [inline src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger]: What if deliveriesToInsert.size() is never greater than 90? What if there are only 5 deliveries? Then they will never be inserted, it seems.

**Comment by jgomezr** [comment]
Ok luke, fix consistent indentation, and up code coverage to 87%

**Comment by jgomezr** [comment]
fix indentation

File: src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+			delivery.Delivery_Date__c = foa.Delivery_Date__c;
+			delivery.Is_Delivery_On_Time__c = 'Sí';
+			delivery.Number_of_Kilos__c = foa.Number_of_Kilos__c;
+		    delivery.Number_of_Units_of_Presentation__c = foa.Number_of_Units_of_Presentation__c;
+		    delivery.Presentacion__c = foa.Presentacion__c;
+		    deliveriesToInsert.add(delivery);
+		    }
+	    }
+	}
+    if (deliveriesToInsert.size() > 90) {
```
Comment (lc0rp) [inline src/triggers/AlsurCreateFarmerProductDeliveryComplete.trigger]: This still will not work if the deliveries are less than 90

**Comment by jgomezr** [comment]
luke,

Fix identation a improve the test

**Comment by jgomezr** [comment]
Hi all,

make changes on code identation and insert adicional update to trigger

**Comment by jgomezr** [comment]
make changes on code identation and insert adicional update to trigger

**Pull request merged**